store/src/java/com/zimbra/cs/ephemeral/EphemeralStore.java

```diff
@@ -234,31 +246,41 @@ public static void setAutoloadExtensions(boolean bool) {
         autoloadExtensions = bool;
     }
 
-    private static String getFactoryClassName(String backendName) throws ServiceException {
```

Comment:
The reason it tries to load the extension if it's not registered is because non-mailboxd processes may not have the proper extension initialized when the backend URL is changed. For example, if imapd is started when LDAP is the backend, and then the backend is changed to SSDB, the SSDB extension may not be loaded since only the current backend was being initialized. The alternative is to load all ephemeral backend extensions preemptively, which is possible, but requires updating all CLI utilities that may touch authentication, and isn't really future-proof. Autoloading just seemed like an easier solution.

Regarding the boolean field: think I can actually remove the static autoloadExtensions field. I recently introduced it in order to disable unnecessary extension loading by ProvUtil and other scripts, but as part of this PR I also silenced those log messages when calling initEphemeralBackendExtension from getFactoryClassName.